webapp/loggedin/review/reducer.js

```diff
@@ -10,30 +8,35 @@ import {
   issueGetSummaryCompleted
 } from './actions'
 
+import * as ReviewState from '@webapp/loggedin/review/reviewState'
+
 const actionHandlers = {
-  [issuePostCommentCompleted]: (state, action) => ({ ...state, 'status': action.status }),
-  [issueRetrieveCommentsStarted]: (state, action) => ({ ...state, 'status': action.status }),
-  [issueRetrieveCommentsCompleted]: (state, action) => ({
-    ...state,
-    [action.target]: R.merge(state[action.target], { issue: action.issue })
-  }),
-  [issueGetSummaryCompleted]: (state, action) =>
-    ({
-      ...state,
-      [action.target]: R.merge(state[action.target], {
-        issuesCount: action.issuesCount,
-        lastCommentUserId: action.lastCommentUserId,
-        issueStatus: action.issueStatus,
-        hasUnreadIssues: action.hasUnreadIssues
-      })
-    })
-  ,
-  [issueOpenCommentThread]: (state, action) => ({
-    ...state,
-    'openThread': { target: action.target, section: action.section, name: action.title }
-  }),
-  [issueCloseCommentThread]: state =>
-    R.omit(['openThread'], state),
+  [issuePostCommentCompleted]: (state, { status }) => ReviewState.assocStatus(status)(state),
+  [issueRetrieveCommentsStarted]: (state, { status }) => ReviewState.assocStatus(status)(state),
+
+  [issueRetrieveCommentsCompleted]: (state, { target, issue }) =>
+    ReviewState.assocDynamicTarget(target)(
+      {
+        ...ReviewState.getDynamicTarget(target),
+        issue
+      }
+    )(state),
```

Comment:
from what i see, the target should be merged . assocDynamicTarget accepts only one target